caikit/runtime/service_factory.py

```diff
@@ -130,9 +136,27 @@ def get_service_package(cls, service_type: ServiceType) -> ServicePackage:
         # Then do API introspection to come up with all the API definitions to support
         caikit_config = get_config()
         clean_modules = ServicePackageFactory._get_and_filter_modules(
-            caikit_config, caikit_config.runtime.library
+            caikit_config, caikit_config.runtime.library, write_modules_file
         )
 
+        # Assert for backwards compatibility, if enabled
+        backwards_compat_conf = (
+            caikit_config.runtime.service_generation.backwards_compatibility
+        )
+        if backwards_compat_conf and backwards_compat_conf.enabled:
+            previous_included_modules = set()
+            prev_modules_path = backwards_compat_conf.prev_modules_path
+            if prev_modules_path:
+                with open(prev_modules_path, "r", encoding="utf-8") as f:
+                    previous_modules = json.load(f)
+                    previous_included_task_map = previous_modules["included_modules"]
+                    for task_module in previous_included_task_map.values():
+                        previous_included_modules.update(task_module.values())
```

Comment:
🤔  Do we want to check by `guid` instead of by name? (`task_module.keys()` instead here)

Checking by name doesn't allow module authors to refactor their module naming or packaging. I'd argue that both of those are `python` API breaking changes, but I don't think they're `runtime` API breaking changes since the runtime apis don't actually care about the name of the module (for inference, at least) nor where it is packaged within the library.

Reply:
Addressed to use `MODULE_ID`